src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -170,30 +103,58 @@ private function createRelationsForItem($item, $formattedRelations)
             if (! isset($relationDetails['model'])) {
                 continue;
             }
-            $model = $relationDetails['model'];
             $relation = $item->{$relationMethod}();
+            $relationType = $relationDetails['relation_type'];
+
+            switch ($relationType) {
+                case 'BelongsTo':
+                    $modelInstance = $relationDetails['model']::find($relationDetails['values'])->first();
+                    if ($modelInstance != null) {
+                        $relation->associate($modelInstance)->save();
+                    } else {
+                        $relation->dissociate()->save();
+                    }
+                    break;
+                case 'HasOne':
+                case 'MorphOne':
+                        $modelInstance = $this->createUpdateOrDeleteOneToOneRelation($relation, $relationMethod, $relationDetails);
+                    break;
+                case 'HasMany':
+                case 'MorphMany':
+                    $relationValues = $relationDetails['values'][$relationMethod];
+                    // if relation values are null we can only attach, also we check if we sent
+                    // - a single dimensional array: [1,2,3]
+                    // - an array of arrays: [[1][2][3]]
+                    // if is as single dimensional array we can only attach.
+                    if ($relationValues === null || ! is_multidimensional_array($relationValues)) {
+                        $this->attachManyRelation($item, $relation, $relationDetails, $relationValues);
+                    } else {
+                        $this->createManyEntries($item, $relation, $relationMethod, $relationDetails);
+                    }
+                    break;
+                case 'BelongsToMany':
+                case 'MorphToMany':
+                    $values = $relationDetails['values'][$relationMethod] ?? [];
+                    $values = is_string($values) ? json_decode($values, true) : $values;
+
+                    $relationValues = [];
+
+                    if (is_multidimensional_array($values)) {
```

Comment:
Technically it does not break, because in backpack we accept:

```php
[] // empty array

[1,2,3] // single dimension array (select)

[ // multidimensional array (repeatable)
    ['key' => 'value'],
    ['key' => 'value']
]
```

We were using both conditions: `isset($subfields)` and `count() vs count(,RECURSIVE)` for multidimensionality. If subfields are not set but the array we receive is multidimensional (because that is the condition to be meet, we don't use `$subfields` for anything, it was just another check) we parse it. 

I don't mind having the aditional check for subfields, but checking for array dimension should be ok too.

Reply:
Ok great, just checking!